compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -944,6 +944,47 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         interp_ok(())
     }
 
+    pub fn prepare_for_native_call(
+        &mut self,
+        id: AllocId,
+        initial_prov: M::Provenance,
+    ) -> InterpResult<'tcx> {
+        // Expose provenance of the root allocation.
+        M::expose_provenance(self, initial_prov)?;
+
+        let mut done = rustc_data_structures::fx::FxHashSet::default();
```

Comment:
```suggestion
        let mut done = FxHashSet::default();
```
Please add `use rustc_data_structures::fx::FxHashSet` at the top of the file